pkg/oc/cli/cmd/process.go

```diff
@@ -129,84 +160,119 @@ func RunProcess(f kcmdutil.Factory, in io.Reader, out, errout io.Writer, cmd *co
 		}
 	}
 
-	local := kcmdutil.GetFlagBool(cmd, "local")
 	if cmd.Flag("param").Changed {
 		flagValues := getFlagStringArray(cmd, "param")
-		cmdutil.WarnAboutCommaSeparation(errout, flagValues, "--param")
+		cmdutil.WarnAboutCommaSeparation(o.ErrOut, flagValues, "--param")
 		templateParams = append(templateParams, flagValues...)
 	}
 
-	duplicatedKeys := sets.NewString()
-	params, paramErr := app.ParseAndCombineEnvironment(templateParams, getFlagStringArray(cmd, "param-file"), in, func(key, file string) error {
-		if file == "" {
-			duplicatedKeys.Insert(key)
-		} else {
-			fmt.Fprintf(errout, "warning: Template parameter %q already defined, ignoring value from file %q", key, file)
+	o.templateParams = templateParams
+	o.templateName = templateName
+
+	o.builderFn = f.NewBuilder
+	o.clientConfigFn = f.ToRESTConfig
+
+	var err error
+	o.mapper, err = f.ToRESTMapper()
+	if err != nil {
+		return err
+	}
+
+	o.namespace, o.explicitNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	// we only need to fail on namespace acquisition if we're actually taking action.  Otherwise the namespace can be enforced later
+	if err != nil && !o.local {
+		return err
+	}
+
+	o.outputFormat = kcmdutil.GetFlagString(cmd, "output")
+
+	if o.outputFormat != "describe" {
```

Comment:
issue to rewrite this into a new printeroptions thing

Reply:
https://github.com/openshift/origin/issues/20316